### PR TITLE
Add support to exclude files and directories.

### LINKED ISSUE #11: Add option to ignore files/folders
Similar to the `--exclude` option in [black](https://github.com/psf/black#configuration) or [autoflake](https://github.com/myint/autoflake#advanced-usage) it would be good to have an option to exclude files and folders

### PR DESCRIPTION
**Pull request by sodul**
Fixes #11

* Add support to exclude files and folders
  Potential backward compatibility issue with the default exclusion list.
  Some files and folders might now be excluded while they would have been
  processed in older versions.
* remove travis.yml since it is no longer usable.

### CODE REVIEW

File: tests/test_processing.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+"""Unittests for gray.processing."""
+
+import pathlib
+import re
+import unittest
+
+from gray import main
+from gray import processing
```
Comment (dizballanze) [inline tests/test_processing.py:8]: `from gray import processing, main`

File: tests/test_processing.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+"""Unittests for gray.processing."""
```
Comment (dizballanze) [inline tests/test_processing.py:1]: Rewrite tests in pytest, please. That's not very convenient to have both unittest and pytest in the same project.

File: gray/processing.py [hunk 3]
```diff
@@ -11,34 +14,102 @@
...
+    Returns:
+        True if the path matches .
+    """
+    if not excludes:
+        return False
+    posix_path = str(path.as_posix())
+    for exclude in excludes:
+        if exclude.match(posix_path):
+            return True
+    return False
```
Comment (dizballanze) [inline gray/processing.py:70]: return any(exclude.match(posix_path) for exclude in excludes)

File: gray/processing.py [hunk 4]
```diff
@@ -11,34 +14,102 @@
 log = logging.getLogger(__name__)
 
 
-class FormattingError(Exception):
+class Error(Exception):
```
Comment (dizballanze) [inline gray/processing.py:17]: `GrayError`, maybe? `Error` is a too broad name.

File: .travis.yml [hunk 5]
```diff
@@ -1,21 +0,0 @@
-# Config file for automatic testing at travis-ci.org
```
Comment (dizballanze) [inline .travis.yml:1]: Thanks!

File: gray/processing.py (hunk 3 above)
Comment (sodul) [reply gray/processing.py:70]: Done

File: gray/processing.py (hunk 4 above)
Comment (sodul) [reply gray/processing.py:17]: Done

File: .travis.yml (hunk 5 above)
Comment (sodul) [reply .travis.yml:1]: You are welcome!

**Review by dizballanze — CHANGES_REQUESTED** [review]

File: tests/test_processing.py (hunk 2 above)
Comment (dizballanze) [reply tests/test_processing.py:1]: Yep, `unittest.mock` is totally fine

File: gray/main.py [hunk 6]
```diff
@@ -92,7 +92,7 @@ def get_parser() -> configargparse.ArgumentParser:
     group.add_argument(
         "-f",
         "--formatters",
-        help=f"Enabled formatters separated by comma"
+        help="Enabled formatters separated by comma"
             " (optional: {OPTIONAL_FORMATTERS_NAMES})",
```
[the remark this answers is not in the document]
Comment (sodul) [reply gray/main.py]: Ah sorry about that, I misplaced it for the new `--exclude` as well.
In practice only the segment with variables need the `f` notation. IIRC correctly newer versions of Pylint will complain about the useless extra notations for example.

I've rerun `gray -h` and updated the README.md file accordingly. Sorry about this oversight.

**Review by dizballanze — APPROVED** [review]

**Comment by dizballanze** [comment]
Released in [0.12.0](https://pypi.org/project/gray/0.12.0/)
Thanks!

**Pull request merged**